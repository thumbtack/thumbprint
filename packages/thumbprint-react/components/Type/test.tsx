import React from 'react';
import { mount } from 'enzyme';
import { Title, Text } from './index';

describe('Title', () => {
    test('renders content that is passed in', () => {
        const wrapper = mount(<Title size={1}>goose</Title>);
        expect(wrapper.text()).toBe('goose');
        expect(wrapper).toMatchSnapshot();
    });

    test('allows size to be customised', () => {
        const wrapper = mount(<Title size={2}>goose</Title>);
        expect(wrapper.find('div').hasClass('title2')).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });

    test('renders a div by default', () => {
        const wrapper = mount(<Title size={1}>goose</Title>);
        expect(wrapper.find('div')).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    test('supports `className`', () => {
        const wrapper = mount(
            <Title size={1} className="white black blue">
                goose
            </Title>,
        );
        expect(wrapper.find('div').hasClass('white black blue')).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });

    test('supports different heading levels', () => {
        const wrapper = mount(
            <Title size={1} headingLevel={6}>
                goose
            </Title>,
        );
        expect(wrapper.find('h6')).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    test('supports `dataTest` prop', () => {
        const wrapper = mount(
            <Title dataTest="Duck" headingLevel={1} size={1}>
                Goose
            </Title>,
        );
        expect(wrapper.find('h1').prop('data-test')).toEqual('Duck');
        expect(wrapper).toMatchSnapshot();
    });

    test('supports `id` prop', () => {
        const wrapper = mount(
            <Title id="Duck" headingLevel={1} size={1}>
                Goose
            </Title>,
        );
        expect(wrapper.find('h1').prop('id')).toEqual('Duck');
        expect(wrapper).toMatchSnapshot();
    });
});

describe('Text', () => {
    test('renders content that is passed in', () => {
        const wrapper = mount(<Text>goose</Text>);
        expect(wrapper.text()).toBe('goose');
        expect(wrapper).toMatchSnapshot();
    });

    test('renders a paragraph by default', () => {
        const wrapper = mount(<Text>goose</Text>);
        expect(wrapper.find('p')).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    test('allows size to be customised', () => {
        const wrapper = mount(<Text size={2}>goose</Text>);
        expect(wrapper.find('p').hasClass('text2')).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });

    test('supports `className`', () => {
        const wrapper = mount(<Text className="white black blue">goose</Text>);
        expect(wrapper.find('p').hasClass('white black blue')).toBe(true);
        expect(wrapper).toMatchSnapshot();
    });

    test('can render a span', () => {
        const wrapper = mount(<Text elementName="span">goose</Text>);
        expect(wrapper.find('span')).toHaveLength(1);
        expect(wrapper).toMatchSnapshot();
    });

    test('supports `dataTest` prop', () => {
        const wrapper = mount(
            <Text dataTest="Duck" size={1}>
                Goose
            </Text>,
        );
        expect(wrapper.find('p').prop('data-test')).toEqual('Duck');
        expect(wrapper).toMatchSnapshot();
    });
});
