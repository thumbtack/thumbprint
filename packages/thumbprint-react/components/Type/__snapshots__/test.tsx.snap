// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text allows size to be customised 1`] = `
<Text
  size={2}
>
  <p
    className="text2"
  >
    goose
  </p>
</Text>
`;

exports[`Text can render a span 1`] = `
<Text
  elementName="span"
>
  <span
    className="text1"
  >
    goose
  </span>
</Text>
`;

exports[`Text renders a paragraph by default 1`] = `
<Text>
  <p
    className="text1"
  >
    goose
  </p>
</Text>
`;

exports[`Text renders content that is passed in 1`] = `
<Text>
  <p
    className="text1"
  >
    goose
  </p>
</Text>
`;

exports[`Text supports \`className\` 1`] = `
<Text
  className="white black blue"
>
  <p
    className="text1 white black blue"
  >
    goose
  </p>
</Text>
`;

exports[`Text supports \`dataTest\` prop 1`] = `
<Text
  dataTest="Duck"
  size={1}
>
  <p
    className="text1"
    data-test="Duck"
  >
    Goose
  </p>
</Text>
`;

exports[`Title allows size to be customised 1`] = `
<Title
  size={2}
>
  <div
    className="title2"
  >
    goose
  </div>
</Title>
`;

exports[`Title renders a div by default 1`] = `
<Title
  size={1}
>
  <div
    className="title1"
  >
    goose
  </div>
</Title>
`;

exports[`Title renders content that is passed in 1`] = `
<Title
  size={1}
>
  <div
    className="title1"
  >
    goose
  </div>
</Title>
`;

exports[`Title supports \`className\` 1`] = `
<Title
  className="white black blue"
  size={1}
>
  <div
    className="title1 white black blue"
  >
    goose
  </div>
</Title>
`;

exports[`Title supports \`dataTest\` prop 1`] = `
<Title
  dataTest="Duck"
  headingLevel={1}
  size={1}
>
  <h1
    className="title1"
    data-test="Duck"
  >
    Goose
  </h1>
</Title>
`;

exports[`Title supports \`id\` prop 1`] = `
<Title
  headingLevel={1}
  id="Duck"
  size={1}
>
  <h1
    className="title1"
    id="Duck"
  >
    Goose
  </h1>
</Title>
`;

exports[`Title supports different heading levels 1`] = `
<Title
  headingLevel={6}
  size={1}
>
  <h6
    className="title1"
  >
    goose
  </h6>
</Title>
`;
